docs/notebooks/JAX_for_the_impatient.ipynb

```diff
@@ -0,0 +1,1293 @@
+{
```

Comment:
<p>I would split line 14 to two lines:</p><pre class="ql-syntax" spellcheck="false">y_pred = jnp.dot(W, x) + b
return jnp.inner(y - y_pred, y - y_pred) / 2.0
</pre><p>(any why <code>jnp.inner</code> instead of <code>jnp.dot</code>?)</p><p> </p><p>Or perhaps better:</p><p> </p><pre class="ql-syntax" spellcheck="false">y_pred = make_predict(W, b)(x)
return jnp.inner(y - y_pred, y - y_pred) / 2.0
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google/flax/pull/542/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='45'/>

Reply:
<p>Hm I don't remember why I ended up not using the make_predict (which was the intent intially). As for inner in fact i don't know, it was kind of my first idea as dot doesn't work in higher dimensions. But reading your question in fact inner doesn't work either. So I don't know :D. In 1D at least it's the same.</p>